### PR TITLE
Added default schema validator.

### PR DESCRIPTION
**Pull request by Hyperkid123**
Validating default schema before rendering Form component.

It will render error warning if validation fails.

screenshot-localhost-8080-2019 01 15-09-44-14

### What is tested
- Is schema object?
- Does schema have fields property?
- Does field have component property?
- Is component listed in component mapper?
- Is component valid React component
- Does component have name | key | title property?
- Is condition in correct format
- Is validate in correct format

### CODE REVIEW

File: src/parsers/default-schema-validator.js [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+
+  if (!Array.isArray(obj.fields)) {
+    throw new DefaultSchemaError(`Component of type ${objectKey} must contain "Fields" property of type array, received type: ${typeof obj.fields}!`);
+  }
+};
+
+const checkCondition = (condition, fieldName) => {
+  if (Array.isArray(condition) || typeof condition !== 'object') {
+    throw new DefaultSchemaError(`
+      Error occured in field defintion with name: "${fieldName}".
```
Comment (rvsia) [inline src/parsers/default-schema-validator.js]: typo: definition (same below)

File: src/parsers/default-schema-validator.js [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+      Field condition must be an object, received ${Array.isArray(condition) ? 'array' : typeof condition}!
+    `);
+  }
+
+  if (!condition.hasOwnProperty('when')) {
+    throw new DefaultSchemaError(`
+      Error occured in field defintion with name: "${fieldName}".
```
Comment (rvsia) [inline src/parsers/default-schema-validator.js]: with name = named, called or with name property ?

File: src/parsers/default-schema-validator.js [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+      Field validate property must be an Array, ${typeof validate} received!
+    `);
+  }
+
+  validate.forEach((validator, index) => {
+    if (Array.isArray(validator) || typeof validator !== 'object') {
+      throw new DefaultSchemaError(`
+        Error occured in field defintion with name: "${fieldName}".
+        Field validator at index: ${index} must be and object, ${Array.isArray(validator) ? 'array' : typeof validator} received!
```
Comment (rvsia) [inline src/parsers/default-schema-validator.js]: an object

File: src/parsers/default-schema-validator.js (hunk 2 above)
Comment (Hyperkid123) [reply src/parsers/default-schema-validator.js]: name refers to the `name` attribute of the JSON

Comment (rvsia) [reply src/parsers/default-schema-validator.js]: No, I just thought it would be sound better with one of that option. 😄

**Pull request merged**